src/System.Data.Common/tests/System/Data/Common/RowUpdatedEventArgsTest.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Xunit;
+
+namespace System.Data.Common
+{
+    public class RowUpdatedEventArgsTest
+    {
+        [Fact]
+        public void ConstructorFail()
+        {
+            ArgumentOutOfRangeException ex = Assert.Throws<ArgumentOutOfRangeException>(() => new RowUpdatedEventArgs(null, null, (StatementType)100, null));
+            Assert.Equal(typeof(ArgumentOutOfRangeException), ex.GetType());
+            Assert.Null(ex.InnerException);
+            Assert.True(ex.Message.IndexOf(nameof(StatementType)) != -1);
+            Assert.True(ex.Message.IndexOf("100") != -1);
+            Assert.Equal(ex.ParamName, nameof(StatementType));
```

Comment:
Rather than using Assert.Equal for the ParamName, you can use the Assert.Throws overload that takes the param name.  But we actually don't use that in the corefx repo because some builds optimize away that information, so instead we use the AssertExtensions.Throws helper, e.g.
```C#
AssertExtensions.Throws<ArgumentOutOfRangeException>(nameof(StatementType), () => new RowUpdatedEventArgs(null, null, (StatementType)100, null));
```

Reply:
Good to know!

Hm.. are the assertions before that (e.g. `ex.Message.IndexOf(nameof(StatementType)) != -1`) necessary at all? I read through some other test cases when writing these ones, so I'm not sure if they're _really_ needed. It seems kind of overkill.

Reply:
> are the assertions before that (e.g. ex.Message.IndexOf(nameof(StatementType)) != -1) necessary at all?

We generally haven't added such validation.